specs/beacon-chain.md

```diff
@@ -324,36 +324,39 @@ def shuffle(values: List[Any],
     """
     values_count = len(values)
 
-    # entropy is consumed in 3 byte chunks
-    # sample_max is defined to remove the modulo bias from this entropy source
-    sample_max = 2 ** 24
-    assert values_count <= sample_max
```

Comment:
I'm a bit scared by `assert`s in consensus code. What happens if the `assert` is hit? Does the beacon chain error out and stall?

Reply:
I'll leave this one for @djrtwo or @hwwhww.

Reply:
The shuffling algorithm is independent of what it is shuffling. It just so happens that it will often (always?) be used to shuffle validators. I think it makes sense to restrict the number of validators somewhere else in the code (either implicitly via deposit size or explicitly with a constant)

This assertion in practice should be handled independently for each client. Assertions in python are bad practice because they can trivially be turned off at runtime.